micro-rdk/src/common/data_manager.rs

```diff
@@ -262,43 +252,128 @@ where
     async fn run<'b>(&mut self, app_client: &'b AppClient) -> Result<(), AppClientError> {
         for collector_key in self.resource_method_keys.iter() {
             let store_lock = self.store.lock().await;
-            let mut reader = match store_lock.get_reader(collector_key) {
-                Ok(reader) => reader,
-                Err(err) => {
-                    log::error!(
-                        "error acquiring reader for collector key ({:?}): {:?}",
-                        collector_key,
-                        err
-                    );
-                    continue;
-                }
-            };
-            let data = match read_messages_for_collector(&mut reader) {
-                Ok(data) => data,
-                Err(err) => {
+            loop {
+                let mut reader = match store_lock.get_reader(collector_key) {
+                    Ok(reader) => reader,
+                    Err(err) => {
+                        log::error!(
+                            "error acquiring reader for collector key ({:?}): {:?}",
+                            collector_key,
+                            err
+                        );
+                        continue;
+                    }
+                };
+                let mut current_chunk: Vec<BytesMut> = vec![];
+                let next_message = match reader.read_next_message() {
+                    Ok(msg) => msg,
+                    Err(err) => {
+                        log::error!(
+                            "error reading message from store for collector key ({:?}): {:?}",
+                            collector_key,
+                            err
+                        );
+                        // we don't want to panic, and creating an AppClientError variant for this case
+                        // feels too specific, so we'll return to exit the function early
+                        break;
```

Comment:
Comment says "return", code says "break", but it looks to me like it actually moves to the next collector.